### ISSUE TITLE
How does Array.prototype.resize interfere with this plugin?

### ISSUE THREAD

**Issue by Chinese1904** [collab-project/videojs-wavesurfer#26]
## Description
I use this snippet to modify one of my arrays.
```javascript
Array.prototype.resize = function(newSize, defaultValue) {
    while(newSize > this.length)
           this.push(defaultValue);
    this.length = newSize;
};
```

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.

Just implement this and you will receive this error message 

### Expected
Please describe what you expected to see.
How is this affecting the wavesurfer? I can't find the problem :(

### Actual
Please describe what actually happened.

### Error output
If there are any errors at all, please include them here.
```
VIDEOJS: ERROR: TypeError: uiElements[element].el is not a function
    at constructor.setupUI (videojs.wavesurfer.js:128)
    at HTMLDivElement.e (video.min.js:18)
    at HTMLDivElement.e (video.min.js:18)
    at HTMLDivElement.d.dispatcher.d.dispatcher (video.min.js:18)
    at Object.k [as trigger] (video.min.js:18)
    at b.a.trigger (video.min.js:12)
    at b.<anonymous> (video.min.js:12)
    at e (video.min.js:18)
c.logByType @ video.min.js:18
i.error @ video.min.js:18
d.dispatcher.d.dispatcher @ video.min.js:18
k @ video.min.js:18
a.trigger @ video.min.js:12
(anonymous) @ video.min.js:12
e @ video.min.js:18
```

## Additional Information
Please include any additional information necessary here. Including the following:
#### videojs/wavesurfer
what version of video.js and wavesurfer.js does this occur with?
5.19.1 and 1.3.7
#### browsers
what browser are affected?
Chrome 57, Firefox 52 and Safari
#### OSes
MacOS
what platforms (operating systems and devices) are affected?

**Comment by thijstriemstra** [comment]
can you reproduce it with older versions of video.js, videojs-wavesurfer?

**Comment by Chinese1904** [comment]
I am so sorry @thijstriemstra - I don't have that in front of me anymore. But on top of my head I refactored the Array.prototype to:

```javascript
      function resize(arr, newSize, defaultValue) {
            while(newSize > arr.length)
                arr.push(defaultValue);
            arr.length = newSize;
        }
```
This ended up working brilliantly. 👍 
Not sure why though haha

**Comment by alexthewilde** [comment]
I'm facing the very same issue with a custom `Array.prototype.insert` method. I'm using videojs-wavesurfer v1.3.3.

**Comment by thijstriemstra** [comment]
This looks like the same issue found in videojs-record: https://github.com/collab-project/videojs-record/commit/0a857bab000b711dfc48603b7d57e9d33af22776 but I'm not sure.

I made the same change in the master version of this plugin, can you try if that fixes it @alexthewilde?

**Comment by alexthewilde** [comment]
@thijstriemstra thanks for looking into this so quickly.

I've tested your fix and the original error has disappeared. However a new error is being thrown in an infinite loop:

```
wavesurfer.js:2064 Uncaught TypeError: Cannot read property 'canvas' of undefined
    at Object.updateDimensions (wavesurfer.js:2064)
    at Object.updateSize (wavesurfer.js:2014)
    at Object.setWidth (wavesurfer.js:1686)
    at Object.drawPeaks (wavesurfer.js:1605)
    at Object.drawBuffer (wavesurfer.js:347)
    at Object.loadDecodedBuffer (wavesurfer.js:382)
    at Object.reloadBuffer (wavesurfer.microphone.min.js:3)
```

I also verified that it's directly related to the `Array.prototype.insert` method.

**Comment by thijstriemstra** [comment]
this one is weird and I don't see any references to videojs-wavesurfer in this traceback. Can you debug and find out what line is causing this issue @alexthewilde?

**Comment by thijstriemstra** [comment]
And why are you changing the array prototype globally. This should cause all kinds of unexpected issues with other libraries?

**Comment by alexthewilde** [comment]
I actually haven't experienced any other issues, only in combination with videojs-wavesurfer.

I ended up re-factoring the method similar to how @Chinese1904 did it as I couldn't find the cause for the second error message and it's not such a big deal really.

**Comment by thijstriemstra** [comment]
v2 of this plugin is coming up (with wavesurfer.js 2.0 support), if this issue still exists, please reopen this ticket.

**Comment by thijstriemstra** [comment]
Released a new version with a fix but didn't test it properly, it broke the live example: https://collab-project.github.io/videojs-wavesurfer/examples/live.html

**Comment by thijstriemstra** [comment]
Should be fixed with 1.3.5 release.
